File: Data Structures & Algorithms/two-integer-sum-ii/submission-5.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        L = 0
        R = len(numbers)-1

        while(L < R):
           numSum = numbers[L] + numbers[R]
           if numSum == target:
              return [L+1, R+1]
           elif numSum < target:
               L += 1
           else:
               R -= 1
        
        return [-1,-1]

File: Data Structures & Algorithms/two-integer-sum/submission-5.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> prevSeen = new HashMap<Integer, Integer>();
        int[] ans = new int[2];

        for(int i=0; i<nums.length; ++i){
            if(prevSeen.get(target-nums[i]) != null){
                ans[0] = prevSeen.get(target-nums[i]);
                ans[1] = i;
                return ans;
            }
            prevSeen.put(nums[i],i);
        }

        return new int[]{};
    }
}
